### PR TITLE
Initial implementation of v2 gcloud action

### PR DESCRIPTION
**Pull request by craigdbarber**

### CODE REVIEW

**Comment by craigdbarber** [comment]
This diff is intense. Might be easier to view the master of my fork for the current state of things: https://github.com/craigdbarber/github-actions

**Comment by craigdbarber** [comment]
@imjohnbo would very much appreciate your feedback.

File: .github/workflows/workflow.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    strategy:
+      matrix:
+        operating-system: [ubuntu-latest, windows-latest, macos-latest]
+    steps:
+    - uses: actions/checkout@master
+    
+    - name: Set Node.js 10.x
+      uses: actions/setup-node@master
+      with:
+        version: 10.x
```
Comment (imjohnbo) [inline .github/workflows/workflow.yml]: I think this property was updated to `node-version`: 

```suggestion
        node-version: 10.x
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -15,14 +28,19 @@ You generally only need to submit a CLA once, so if you've already submitted one
...
 use GitHub pull requests for this purpose. Consult
 [GitHub Help](https://help.github.com/articles/about-pull-requests/) for more
 information on using pull requests.
 
-## Community Guidelines
+## Style Guide
 
-This project follows [Google's Open Source Community
-Guidelines](https://opensource.google/conduct/).
+This project conforms to the [Google Java Style Guide](https://google.github.io/styleguide/javaguide.html). All submitted PRs will be required to conform to this style guide before being merged.
```
Comment (imjohnbo) [inline CONTRIBUTING.md]:
```suggestion
This project conforms to the [Google JavaScript Style Guide](https://google.github.io/styleguide/jsguide.html). All submitted PRs will be required to conform to this style guide before being merged.
```

**Comment by imjohnbo** [comment]
@googlebot I fixed it.

**Comment by imjohnbo** [comment]
@googlebot I consent.

File: setup-gcloud/src/client-util.ts [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+            objectName = `google-cloud-sdk-${version}-darwin-${arch}.tar.gz`;
+            break;
+        case 'win32':
+            objectName = `google-cloud-sdk-${version}-windows-${arch}.zip`;
+            break;
+        default:
+            throw new Error(`Unexpected OS '${os}'`);
+    }
+
+    return encodeURI(`https://www.googleapis.com/storage/v1/b/cloud-sdk-release/o?prefix=${objectName}`);
```
Comment (evandbrown) [inline setup-gcloud/src/client-util.ts]: Should the URL template be a var or const?

File: setup-gcloud/dist/src/client-util.js [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+        case 'darwin':
+            objectName = `google-cloud-sdk-${version}-darwin-${arch}.tar.gz`;
+            break;
+        case 'win32':
+            objectName = `google-cloud-sdk-${version}-windows-${arch}.zip`;
+            break;
+        default:
+            throw new Error(`Unexpected OS '${os}'`);
+    }
+    return encodeURI(`https://www.googleapis.com/storage/v1/b/cloud-sdk-release/o?prefix=${objectName}`);
```
Comment (evandbrown) [inline setup-gcloud/dist/src/client-util.js]: Should URL template be a var or const?

File: setup-gcloud/README.md [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+1. Downloads a version of the [gcloud SDK](https://cloud.google.com/sdk/) according to the specified `version` input, as well as the environment OS and architecture.
+
+2. Installs and caches the downloaded version into the actions environment.
+
+3. Authenticates the gcloud CLI tool using the inputs: `service_account_email` and `service_account_key`.
+
+## Prerequisites
+
+* This actions requires [Python](https://www.python.org/) 2.7.9 or later to be installed on the environment.
```
[the remark this answers is not in the document]
Comment (craigdbarber) [reply setup-gcloud/README.md]: Done.

File: setup-gcloud/README.md [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+
+ Unless required by applicable law or agreed to in writing, software distributed under the License
+ is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ implied. See the License for the specific language governing permissions and limitations under the
+ License.
+-->
+
+# setup-gcloud
+
+This action configures the [gcloud SDK](https://cloud.google.com/sdk/) in the environment for use in actions.
```
[the remark this answers is not in the document]
Comment (craigdbarber) [reply setup-gcloud/README.md]: Done.

File: setup-gcloud/README.md [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+  with:
+      version: '270.0.0'
+      service_account_email: ${{ secrets.GCP_SA_EMAIL }}
+      service_account_key: ${{ secrets.GCP_SA_KEY }}
+- run: gcloud info
+```
+
+## Inputs
+
+* `version`: The version of the gcloud SDK to be installed.  Example: `270.0.0`
```
[the remark this answers is not in the document]
Comment (craigdbarber) [reply setup-gcloud/README.md]: Done.

File: setup-gcloud/src/client-util.ts (hunk 3 above)
Comment (craigdbarber) [reply setup-gcloud/src/client-util.ts]: If it were used anywhere else I'd agree, but its use is confined to here.

File: setup-gcloud/dist/src/client-util.js (hunk 4 above)
Comment (craigdbarber) [reply setup-gcloud/dist/src/client-util.js]: Everything within dist/ is transpiled code.

File: setup-gcloud/.prettierrc.js [hunk 8]
```diff
@@ -0,0 +1,7 @@
+module.exports = {
```
Comment (averikitsch) [inline setup-gcloud/.prettierrc.js]: DevRel uses https://github.com/GoogleCloudPlatform/nodejs-docs-samples/blob/master/.prettierrc

File: setup-gcloud/README.md [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+ License.
+-->
+
+# setup-gcloud
+
+This action configures the [Google Cloud SDK](https://cloud.google.com/sdk/) in the environment for use in actions.
+
+It does the following:
+
+1. Downloads a version of the [Google Cloud SDK](https://cloud.google.com/sdk/) according to the specified `version` input, as well as the environment OS and architecture.
```
Comment (averikitsch) [inline setup-gcloud/README.md]: It's helpful to mention the Cloud SDK == gcloud

File: setup-gcloud/README.md [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+
+2. Installs and caches the downloaded version into the actions environment.
+
+3. Authenticates the gcloud CLI tool using the inputs: `service_account_email` and `service_account_key`.
```
Comment (averikitsch) [inline setup-gcloud/README.md]: It would be good to link to or add how to get these.

File: setup-gcloud/README.md (hunk 9 above)
Comment (craigdbarber) [reply setup-gcloud/README.md]: Done.

File: setup-gcloud/README.md (hunk 10 above)
Comment (craigdbarber) [reply setup-gcloud/README.md]: Done.

**Pull request merged**